From 89de414c8ecf98ab8fd592efd51231c64b2de54c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:23:28 -0400
Subject: [tools] perf/build: split out feature check 'volatile-register-var'

Message-id: <1408462094-14194-40-git-send-email-jolsa@redhat.com>
Patchwork-id: 88023
O-Subject: [PATCH RHEL7.1 BZ1131394 039/325] tools/perf/build: Split out feature check: 'volatile-register-var'
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit c25104452de94102810e86323eabb7d9a4cad083
Author: Ingo Molnar <mingo@kernel.org>
Date: Mon Sep 30 13:58:12 2013 +0200
---
 create mode 100644 tools/perf/config/feature-checks/test-volatile-register-var.c

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index 9f94912..a3de2f8 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -105,6 +105,7 @@ FEATURE_TESTS =    \
  hello    \
  stackprotector-all  \
  stackprotector   \
+ volatile-register-var  \
  libnuma
 
 $(foreach test,$(FEATURE_TESTS),$(call feature_check,$(test),$(test)))
@@ -117,7 +118,7 @@ ifeq ($(feature-stackprotector), 1)
   CFLAGS += -Wstack-protector
 endif
 
-ifeq ($(call try-cc,$(SOURCE_HELLO),$(CFLAGS) -Werror -Wvolatile-register-var,-Wvolatile-register-var),y)
+ifeq ($(feature-volatile-register-var), 1)
   CFLAGS += -Wvolatile-register-var
 endif
 
diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index 46b7650..5693299 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -3,6 +3,7 @@ FILES=     \
  test-hello   \
  test-stackprotector-all  \
  test-stackprotector  \
+ test-volatile-register-var \
  test-libnuma
 
 CC := $(CC) -MD
@@ -22,6 +23,9 @@ test-stackprotector-all:
 test-stackprotector:
  $(BUILD) -Werror -fstack-protector
 
+test-volatile-register-var:
+ $(BUILD) -Werror -Wvolatile-register-var
+
 test-libnuma:
  $(BUILD) -lnuma
 
diff --git a/tools/perf/config/feature-checks/test-volatile-register-var.c b/tools/perf/config/feature-checks/test-volatile-register-var.c
new file mode 100644
index 0000000..c9f398d
--- /dev/null
+++ b/tools/perf/config/feature-checks/test-volatile-register-var.c
@@ -0,0 +1,6 @@
+#include <stdio.h>
+
+int main(void)
+{
+ return puts("hi");
+}
-- 
1.7.1